crates/redis_interface/src/lib.rs

```diff
@@ -117,6 +172,32 @@ impl RedisConnectionPool {
     }
 }
 
+#[async_trait::async_trait]
+pub trait PubSubInterface {
+    async fn subscribe(&self, channel: &str) -> CustomResult<usize, errors::RedisError>;
+    async fn publish(&self, channel: &str, key: &str) -> CustomResult<usize, errors::RedisError>;
+    async fn on_message(&self);
+}
+
+#[async_trait::async_trait]
+impl PubSubInterface for RedisConnectionPool {
+    #[inline]
+    async fn subscribe(&self, channel: &str) -> CustomResult<usize, errors::RedisError> {
+        self.subscriber.subscribe(channel).await
+    }
+    #[inline]
+    async fn publish(&self, channel: &str, key: &str) -> CustomResult<usize, errors::RedisError> {
+        self.publisher.publish(channel, key).await
+    }
+    #[inline]
+    async fn on_message(&self) {
+        let mut message = self.subscriber.on_message();
+        while let Some((_, key)) = message.next().await {
+            logger::error!("{:?}", key);
+        }
```

Comment:
1. Can we use `for_each()` maybe?
2. Is the key being logged at the `ERROR` level intentional? Maybe include some context as to what's the error here?
   If you're updating this snippet, a minor change: `logger::error!("{key:?}");`

Reply:
that log is just a placeholder I will change it to appropriate behaviour

Reply:
Reopening this conversation since there's nothing being done with `key` as of now:

https://github.com/juspay/hyperswitch/blob/cd1e21b2de00536a6d20457143f7897d3feacdf1/crates/redis_interface/src/lib.rs#L182-L185